hs_core/tests/api/native/test_create_new_version_resource.py

```diff
@@ -0,0 +1,181 @@
+import os
+import tempfile
+import shutil
+
+from django.contrib.auth.models import Group
+from django.test import TestCase
+from django.core.exceptions import ValidationError
+from django.core.files.uploadedfile import UploadedFile
+
+from hs_core import hydroshare
+from hs_core.models import GenericResource
+from hs_core.hydroshare import utils
+from hs_geo_raster_resource.models import RasterResource, OriginalCoverage, CellInformation, BandInformation
+
+class TestNewVersionResource(TestCase):
+    def setUp(self):
+        super(TestNewVersionResource, self).setUp()
+        self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+
+        # create a user who is the owner of the resource to be versioned
+        self.owner = hydroshare.create_account(
+            'owner@gmail.edu',
+            username='owner',
+            first_name='owner_firstname',
+            last_name='owner_lastname',
+            superuser=False,
+            groups=[]
+        )
+
+        # create a user who is NOT the owner of the resource to be versioned
+        self.nonowner = hydroshare.create_account(
+            'nonowner@gmail.com',
+            username='nonowner',
+            first_name='nonowner_firstname',
+            last_name='nonowner_lastname',
+            superuser=False,
+            groups=[]
+        )
+
+        # create a generic resource
+        self.res_generic = hydroshare.create_resource(
+            resource_type='GenericResource',
+            owner=self.owner,
+            title='Test Generic Resource'
+        )
+
+        # create a raster resource that represents a specific resource type
+        raster_file = 'hs_core/tests/data/cea.tif'
+        temp_dir = tempfile.mkdtemp()
+        temp_raster_file = os.path.join(temp_dir, 'cea.tif')
+        shutil.copy(raster_file, temp_raster_file)
+        self.raster_obj = open(temp_raster_file, 'r')
+        files = [UploadedFile(file=self.raster_obj, name='cea.tif')]
+        _, _, metadata = utils.resource_pre_create_actions(resource_type='RasterResource',
+                                                          resource_title='Test Raster Resource',
+                                                          page_redirect_url_key=None,
+                                                          files=files,
+                                                          metadata=None,)
+        self.res_raster = hydroshare.create_resource(
+            resource_type='RasterResource',
+            owner=self.owner,
+            title='Test Raster Resource',
+            files=files,
+            metadata=metadata
+        )
+
+    def test_new_version_generic_resource(self):
```

Comment:
@pkdash I have added test cases to address your second and third bullets. See if they are sufficient. As for the first bullet, I'll defer to @horsburgh for the final words, but in my opinion, we should allow superuser to intervene in cases where things could go wrong, which is why we allow useruser to change resource that has been published, just in case if there is an issue with published resource. In this specific case, there is no intervening or error-correcting role but create a new resource, so I don't see the need for allowing superuser to do that.